frontend/src/elements/SlidingPanel.vue

```diff
@@ -0,0 +1,233 @@
+<script setup lang="ts">
+import { computed } from 'vue'
+import { IconX } from '@tabler/icons-vue'
+
+interface Props {
+  open: boolean
+  title?: string
+  closeOnScrimClick?: boolean
+  side?: 'left' | 'right'
+}
+
+interface Emits {
+  (e: 'update:open', value: boolean): void
+  (e: 'close'): void
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  closeOnScrimClick: true,
+  side: 'right'
+})
+
+const emit = defineEmits<Emits>()
+
+const isOpen = computed({
+  get: () => props.open,
+  set: (value) => emit('update:open', value)
+})
+
+const handleClose = () => {
+  isOpen.value = false
+  emit('close')
+}
+
+const handleScrimClick = () => {
+  if (props.closeOnScrimClick) {
+    handleClose()
+  }
+}
+
+const panelClasses = computed(() => [
+  'sliding-panel',
+  `sliding-panel--${props.side}-side`
+])
+
+const headerClasses = computed(() => [
+  'header',
+  `header--${props.side}-side`
+])
+</script>
+
+<template>
+  <Teleport to="body">
+    <Transition name="scrim">
+      <div
+        v-if="isOpen"
+        class="scrim"
+        @click="handleScrimClick"
+      />
+    </Transition>
+    <Transition :name="`panel-${side}`">
```

Comment:
Can we add a prefers-reduced-motion to remove the slide in effect?
https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion

You can test it by setting `ui.prefersReducedMotion` to `1` in your `about:config`.